INSTALL.adoc

```diff
@@ -198,3 +198,71 @@ and sanity checks that could help you pinpoint the problem.
   requirements for mathematical accuracy, numerical precision, and proper
   handling of mathematical corner cases and error conditions may need to
   consider running their code on a platform with better C99 support.
+
+== (experimental) Building a cross compiler
+
+A cross compiler is a compiler that runs on some machine, named the
+_host_, but generate code for different machine, named the _target_. To
+build a cross compiler you first need to have a non-cross compiler of
+the same version installed in your `$PATH`. Note that they need to be of
+the same version since the cross compilers will contain the runtime of
+the non-cross compiler because the C toolchain that will be detected
+during the `configure` for the OCaml cross compiler will be the C cross
+compiler to `target`. It may be useful to keep a note of the flags used
+to link the non-cross compiler with `zstd`, if any:
+
+....
+grep ZSTD_LIBS Makefile.config
+....
+
+before cleaning up the working directory to build the cross compiler.
+
+To start the build of the cross compiler, call `configure` with the
+`target` triplet, possibly setting where the library will be installed
+on the target. So for instance, with the GCC MinGW cross compiler
+installed, it could be:
+
+....
+./configure --prefix=$PWD/cross --target=x86_64-w64-mingw32 --with-target-libdir='C:\somedir' ...
+make crossopt -j
+make installcross
+....
+
+Notes:
+
+* It is advisable to choose a `prefix` that will not end up in
+installing the cross compiler in your `$PATH`, since you probably want
+to make sure the `ocamlopt` command runs the non-cross compiler, not the
+cross one. When you want to call the cross compiler, you will just call
+it using its full path or add it temporarily to your `$PATH`.
```

Comment:
(consider adding machinery to prefix the binary names with the target triplet, as usually done for C cross compilers, i.e. the cross-compiler for fisherprice-tablet-mos6502 is called `fisherprice-tablet-mos6502-gcc`).

Reply:
I considered that but I followed the current practice in the OCaml cross compilers projects, namely that the build tools (`ocamlfind` and `dune`) rely on `ocamlfind` configuration files to find the cross compilers.
In particular, in a private conversation, @dra27 proposed the idea that a nicer design to aim for could be to allow an OCaml compiler to have more than one target. So these design decisions are left for future work.